iris_grib/__init__.py

```diff
@@ -298,6 +300,12 @@ def _compute_extra_keys(self):
             centre_number=centre_number,
             param_number=self.indicatorOfParameter)
         self.extra_keys['_cf_data'] = cf_data
+        self.extra_keys['_grib_code'] = gptx.GRIBCode(
+            edition_or_string=1,
+            discipline=self.table2Version,
+            category=centre_number,
+            number=self.indicatorOfParameter
+        )
```

Comment:
I can't compare this to the GRIB-2 call to verify its correctness, since the GRIB-2 call is very generic.

https://github.com/SciTools/iris-grib/blob/368596ab76845a61f36b086c07c0a35982f692dd/iris_grib/_load_convert.py#L1449-L1454

Can you point me to anything that will verify these arguments are correct?

Reply:
The GRIBcode is just a NamedTuple, so I decided to just re-use the existing one with its named properties [as described here](https://github.com/SciTools/iris-grib/blob/368596ab76845a61f36b086c07c0a35982f692dd/docs/index.rst?plain=1#L59-L70). 
It's certainly true that its repr is less than ideal for this purpose, but I just thought it was the simplest possible solution.

Idealy, I **_could_** re-configure this to use a different GRIBcode type, which could properly represent GRIB1 usage with different + more suitable named properties.  If you think it's important that this should properly reflect the keys which define a GRIB1 phenomenon, then I certainly still can go that way.  I just decided not to when I realised it will mean changing existing code in quite a few different places.

Reply:
P.S. you might notice that we can't _write_ GRIB1 defined phenomena anyway, since [this code](https://github.com/SciTools/iris-grib/blob/main/iris_grib/_save_rules.py#L784) prevents it.

Reply:
I appreciate that you didn't want to put in more effort than necessary, but I think we need something clearer unfortunately.

I've spent an hour or so reading around the topic and I _think_ what you're saying is there is no significance to these keys and you are just using them as convenient places to store information? But I'm still not sure. If it's confused me it will confuse other people.

Reply:
OK agreed.  I'll figure out a way of making this better.